core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -305,6 +306,24 @@ public SqlToRelConverter(
     shouldCreateValuesRel = true;
     isExplain = false;
     nDynamicParamsInExplain = 0;
+    inSubqueryThreshold = DEFAULT_IN_SUBQUERY_THRESHOLD;
```

Comment:
These properties had better go into a default instance of Config implementation. 

SqlToConverter constructor without config should use the default config instance. 
  this(viewExpander, validator, catalogReader, cluster, convertletTable, default_config)

Move line 289-303 to the constructor with config parameter, which will has the full logic.